src/@utils/pool.ts

```diff
@@ -0,0 +1,87 @@
+import { approve, Config, Pool, ZERO_ADDRESS } from '@oceanprotocol/lib'
+import Web3 from 'web3'
+import { getSiteMetadata } from './siteConfig'
+import { getDummyWeb3 } from './web3'
+import { TransactionReceipt } from 'web3-eth'
+
+export async function calculateBuyPrice(
+  accessDetails: AccessDetails,
+  web3?: Web3,
+  chainId?: number
+): Promise<PriceAndEstimation> {
+  if (web3 && chainId)
+    throw new Error("web3 and chainId can't be undefined at the same time!")
+
+  if (!web3) {
+    web3 = await getDummyWeb3(chainId)
+  }
+
+  const pool = new Pool(web3)
+  const { appConfig } = getSiteMetadata()
+  const priceAndEstimation = {
+    price: '0',
+    gasEstimation: 0
+  } as PriceAndEstimation
+
+  const estimatedPrice = await pool.getAmountInExactOut(
+    accessDetails.addressOrId,
+    accessDetails.baseToken.address,
+    accessDetails.datatoken.address,
+    '1',
+    appConfig.marketFee
+  )
+  priceAndEstimation.price = estimatedPrice
+
+  const estimation = await pool.estSwapExactAmountOut(
+    ZERO_ADDRESS,
+    accessDetails.addressOrId,
+    {
+      marketFeeAddress: appConfig.marketFeeAddress,
+      tokenIn: accessDetails.baseToken.address,
+      tokenOut: accessDetails.datatoken.address
+    },
+    {
+      maxAmountIn: (10 * accessDetails.price).toString(),
+      swapMarketFee: appConfig.marketFee,
+      tokenAmountOut: '1'
+    }
+  )
+  priceAndEstimation.gasEstimation = estimation
+
+  return priceAndEstimation
+}
+
+export async function buyDtFromPool(
+  accessDetails: AccessDetails,
+  accountId: string,
+  config: Config,
+  web3: Web3
+): Promise<TransactionReceipt> {
+  const pool = new Pool(web3)
+  const { appConfig } = getSiteMetadata()
+  await approve(
+    web3,
+    accountId,
+    config.oceanTokenAddress,
```

Comment:
we should limit the usage of getting token info from the ocean config, as what we want to have here is the address of the base token from the pool the user wants to buy a datatoken from. So should rather be `accessDetails.baseToken.address` and then no need to pass `config`